aggregator_core/src/datastore.rs

```diff
@@ -3890,12 +3947,22 @@ impl<C: Clock> Transaction<'_, C> {
     ) -> Result<(), Error> {
         let stmt = self
             .prepare_cached(
-                "INSERT INTO outstanding_batches (
+                "WITH non_gc_batches AS (
+                    SELECT
+                        tasks.task_id, batch_identifier
```

Comment:
It ignores the aggregation parameter because `outstanding_batches` is not aggregation-parameter-aware; `outstanding_batches` is not aggregation-parameter-aware since it is used for `current_batch` queries which only work with batches that have not yet been aggregated (or, more precisely, associated with a collection job).

Added a reference to #225.